File: Protogen/loxim_protocol/cpp/protocol/packages_data/SequencePackage.cpp
#include <string.h>
#include <stdlib.h>
#include "SequencePackage.h"
#include "ptools/PackageBufferWriter.h"
#include "ptools/PackageBufferReader.h"

using namespace protocol;

SequencePackage::SequencePackage()
{
}

SequencePackage::~SequencePackage()
{
}

void SequencePackage::serializeW(PackageBufferWriter *writer)
{
	CollectionPackage::serializeW(writer);
}

bool SequencePackage::deserializeR(PackageBufferReader *reader)
{
	if (!CollectionPackage::deserializeR(reader)) return false;
	return true;
}

bool SequencePackage::equals(Package *_package)
{
	if (!_package) return false;
	if (_package->getPackageType()!=this->getPackageType()) return false;
	if (!CollectionPackage::equals(_package)) return false;
	SequencePackage* p=(SequencePackage*)_package;
	return true;
}

File: Protogen/loxim_protocol/java/src/main/java/pl/edu/mimuw/loxim/protocol/packages_data/SequencePackage.java
package pl.edu.mimuw.loxim.protocol.packages_data;

import java.util.EnumSet;
import pl.edu.mimuw.loxim.protogen.lang.java.template.ptools.Package;
import pl.edu.mimuw.loxim.protogen.lang.java.template.exception.ProtocolException;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageInputStreamReader;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageOutputStreamWriter;


import pl.edu.mimuw.loxim.protocol.enums.*;


public  class SequencePackage extends CollectionPackage {
		public final static long ID=(long)133;
	
    	
	public SequencePackage()
	{
	}

		
		@Override
	public long getPackageType() {
		return (long)ID;
	}
		
	@Override
	protected void deserializeW(PackageInputStreamReader reader)
			throws ProtocolException {
		super.deserializeW(reader);
		
			}
	
	@Override
	protected void serializeW(PackageOutputStreamWriter writer)
			throws ProtocolException {
		super.serializeW(writer);
			}
	
	@Override
	public boolean equals(Object obj) {
		if(!super.equals(obj))
			return false;
		SequencePackage p=(SequencePackage)obj;
	    		return true;
	}
	
//================= GETTERS AND SETTERS ===================

    
	 
}


